src/huggingface_hub/inference/_client.py

```diff
@@ -773,6 +773,40 @@ def text_to_speech(self, text: str, *, model: Optional[str] = None) -> bytes:
         response = self.post(json={"inputs": text}, model=model, task="text-to-speech")
         return response.content
 
+    def zero_shot_image_classification(self, image: ContentT, text: Any, *, model: Optional[str] = None) -> str:
+        """
+        Provide input image and text labels to predict text labels for the image.
+
+        Args:
+            image (`Union[str, Path, bytes, BinaryIO]`):
+                The input image to caption. It can be raw bytes, an image file, or a URL to an online image..
+            text (Any):
+                String or List representation of possible class names.
+            model (`str`, *optional*):
+                The model to use for inference. Can be a model ID hosted on the Hugging Face Hub or a URL to a deployed
+                Inference Endpoint. This parameter overrides the model defined at the instance level. Defaults to None.
+
+        Returns:
+            `str`: Class with highest probability.
+
+        Raises:
+            [`InferenceTimeoutError`]:
+                If the model is unavailable or the request times out.
+            `HTTPError`:
+                If the request fails with an HTTP error status code other than HTTP 503.
+
+        Example:
+        ```py
+        >>> from pathlib import Path
+        >>> from huggingface_hub import InferenceClient
+        >>> client = InferenceClient()
+
+        >>> score = client.zero_shot_image_classification("https://upload.wikimedia.org/wikipedia/commons/thumb/4/43/Cute_dog.jpg/320px-Cute_dog.jpg", "dog,cat,horse", model="openai/clip-vit-base-patch32")
+        ```
+        """
+        response = self.post(data={image, text}, model=model, task="zero-shot-image-classification")
```

Comment:
Looking at the payload sent when trying the widget on a model (see screenshot) and the similar [zero-classification-task](https://huggingface.co/docs/api-inference/detailed_parameters#zeroshot-classification-task) docs, the input format is not correct. It is expected to get a base64-encoded image + some parameters.

```suggestion
        response = self.post(
            json={"image": _b64_encode(image), "parameters": {"candidate_labels": text}},
            model=model,
            task="zero-shot-image-classification",
        )
```

![2023-06-26_14-11](https://github.com/huggingface/huggingface_hub/assets/11801849/57d709b2-7ce0-47e5-adb9-f99a54f4d6a1)